lib/multi_configs.js

```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const path = require('path');
+
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const isPlainObject = require('lodash.isplainobject');
+const mapValues = require('lodash.mapvalues');
+
+const expandTilde = require('expand-tilde');
+const parsePath = require('parse-filepath');
+
+
+function search(options, extensions, cwd) {
+  cwd = path.resolve(isString(cwd) ? cwd : '.');
+  extensions = normalizeExtensions(extensions);
+
+  if (!isPlainObject(options)) {
+    return {};
+  }
+
+  return mapValues(options, function(opt, name) {
+    return firstUnNull(expandEachProp(opt, name), findWithExpandedPaths);
+  });
+
+
+  function expandEachProp(opt, name) {
+    if (isString(opt)) {
+      return createExpandedPathArray(opt, name, extensions, false);
+    }
+
+    if (isPlainObject(opt) && isString(opt.path)) {
+      var exts = normalizeExtensions(opt.extensions, extensions);
+      return createExpandedPathArray(opt.path, name, exts, !!opt.findUp);
+    }
+
+    return [];
+  }
+
+  function createExpandedPathArray(dir, name, exts, findUp) {
+    return exts.map(function(ext) {
+      var fp;
+      if (name === '') {
+        fp = path.join(expandTilde(dir) + ext);
+      } else {
+        fp = path.join(expandTilde(dir), name + ext);
+      }
+      return normalizeExpandedPath({ path: fp, basedir: cwd, findUp: findUp });
+    });
+  }
+}
+
+function normalizeExpandedPath(expanded) {
+  var parsed = parsePath(expanded.path);
+
+  if (parsed.isAbsolute) {
+    expanded.findUp = false;
+    expanded.path = expanded.path.slice(parsed.root.length);
+    expanded.basedir = parsed.root;
+
+  } else if (parsed.root) { // expanded.path has a drive letter on windows.
+    expanded.path = expanded.path.slice(parsed.root.length);
+    expanded.basedir = path.resolve(parsed.root);
+  }
+
+  return expanded;
+}
+
+function findWithExpandedPaths(expanded) {
+  return expanded.findUp ?
+    findup(expanded.path, expanded.basedir) :
+    find(path.resolve(expanded.basedir, expanded.path));
+}
+
+
+function normalizeExtensions(exts, def) {
+  if (isString(exts)) {
+    return [exts];
+  }
+
+  if (isArray(exts)) {
+    exts = exts.filter(isString);
+    return (exts.length > 0) ? exts : [''];
+  }
+
+  if (isPlainObject(exts)) {
+    exts = Object.keys(exts);
+    return (exts.length > 0) ? exts : [''];
+  }
+
+  return def || [''];
+}
+
+function find(filepath) {
```

Comment:
Why do you need to check it is a file?  We should be able to resolve directories too because `require('somedir')` will still load, given their is an `index.js` file or a `package.json` describing the main file.

We should only care that something exists; whether it is a file, directory, symlink should be irrelevant to us.